### PR TITLE
Move top-level .gitignore entries to their closest subdirectory

### LINKED ISSUE #20263: Push top-level .gitignore entries into directory-specific .gitignores
This is a thankless housekeeping task (except that I will thank you if I take it on:
Early in the project, for whatever reason, we were putting all .gitignore entries
into a top-level file in `$CHPL_HOME`.  As the project has grown and matured,
this has seemed increasingly silly to me from a maintenance perspective since
putting the entries into directory-specific .gitignore files keeps the information
more localized to where it's useful, making it simpler to maintain as directories
get renamed, moved, or deleted.  When talking about this change in philosophy
with team members, the reaction has generally been positive.

Unfortunately, as long as our top-level .gitignore file continues to have entries,
it continues to accumulate additional entries.  This task proposes that we move
all entries into their closest subdirectory to break the cycle and make it less
ambiguous where we want the entries to go.

### PR DESCRIPTION
**Pull request by Guillaume-Helbecque**
As suggested by #20263, this PR moves most of top-level `.gitignore` entries into their closest subdirectory. It consists in copy/pasting those entries into new local `.gitignore` files, taking care of adjusting file-paths.

The following top-level entries have been removed since their directory (in bold) doesn't exist:

/test/compflags/**lydia**/library/libtesting.a
/test/expressions/lydia/**noinit**/ccode
/test/library/standard/FileSystem/**chmod**/dir
/test/library/standard/FileSystem/**chmod**/file
/test/library/standard/FileSystem/**chmod**/chmod_noprivilege.good
/test/library/standard/Path/**saru**/getParentName/parentName.good
/test/library/standard/Path/**saru**/getParentName/parentNameErr.good
/test/studies/ssca2/**sungeun**/SSCA2_main.RMAT.good
/test/studies/ssca2/**sungeun**/SSCA2_main.1Dtorus.good
/test/studies/ssca2/**sungeun**/SSCA2_main.2Dtorus.good
/test/studies/ssca2/**sungeun**/SSCA2_main.3Dtorus.good
/test/studies/ssca2/**sungeun**/SSCA2_main.4Dtorus.good
/test/studies/lulesh/**sungeun**/coords.out
/test/variables/**bradc**/docs/pragmaMultipleVars.doc.txt

Resolve #20263

[Reviewed by ]

### CODE REVIEW

**Comment by Guillaume-Helbecque** [comment]
Hello @bradcray and @lydia-duncan, I hope this PR fixes #20263 as you'd expected. 

I just had one question when making it: as mentioned above, I didn't find the directory of some top-level `.gitignore` entries. I wonder if this is normal or if we can remove them (as I did)?

(This is my first ever PR so please review it carefully.)

**Comment by lydia-duncan** [comment]
Thanks for taking this on!  I agree, removing the entries that no longer exist is the right call

**Review by lydia-duncan — APPROVED** [review]
This is great and incredibly helpful, thank you!  I just had one comment about the ones in the new compiler .gitignore, but otherwise I think this is in good shape!

I'll go ahead and check out your branch locally to make sure we didn't miss anything (but I don't expect anything to turn up)

File: compiler/.gitignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Compiler ignores.
+/BROWSE
+/chpl
+/ifa/prim_data.cpp
+/ifa/prim_data.h
+/ifa/make_prims/make_prims
+/ifa/make_prims/cygwin
```
[the remark this answers is not in the document]
Comment (Guillaume-Helbecque) [reply compiler/.gitignore]: Done. Thank you for the review!

**Comment by bradcray** [comment]
Thanks for picking this up, @Guillaume-Helbecque !

**Pull request merged**